jerry-core/vm/vm.c

```diff
@@ -1240,7 +1238,80 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
 
           byte_code_p = byte_code_start_p + 1;
           result = last_completion_value;
-          result_p = ecma_get_number_from_value (result);
+
+          if (ecma_is_value_integer_number (result))
+          {
+            ecma_integer_value_t int_value = (ecma_integer_value_t) result;
+            ecma_integer_value_t int_increase;
+
+            if (base & 0x2)
+            {
+              /* For decrement operators */
+              if (int_value <= (ECMA_INTEGER_NUMBER_MIN << ECMA_DIRECT_SHIFT))
+              {
+                int_increase = 0;
+              }
+              else
+              {
+                int_increase = -(1 << ECMA_DIRECT_SHIFT);
+              }
+            }
+            else
+            {
+              if (int_value >= (ECMA_INTEGER_NUMBER_MAX << ECMA_DIRECT_SHIFT))
+              {
+                int_increase = 0;
+              }
+              else
+              {
+                int_increase = 1 << ECMA_DIRECT_SHIFT;
+              }
+            }
+
+            if (int_increase != 0)
+            {
+              /* Post operators require the unmodifed number value. */
+              if (base & 0x4)
+              {
+                if (opcode_data & VM_OC_PUT_STACK)
+                {
+                  if (base & 0x1)
```

Comment:
Please, make an issue for this. This is something we should definitely address. Hard-coded constants (flags, bit masks, etc.) should be removed from the code base.